contracts/tokenWhitelist.sol

```diff
@@ -152,7 +177,91 @@ contract TokenWhitelist is ENSResolvable, Controllable, Transferrable {
         }
     }
 
-    /// @notice Toggles whether or not a token is loadable or not
+    /// @notice Add method signatures(methodIDs) to the whitelist.
+    /// @param _methods method signatures to be added.
+    function addMethodIds(bytes4[] calldata _methods) external onlyAdmin {
+        for (uint i = 0; i < _methods.length; i++) {
+            bytes4 method = _methods[i];
+            if (!_methodIdWhitelist[method]) {
+                _methodIdWhitelist[method] = true;
+                emit AddedMethodId(method);
+            }
+        }
+    }
+
+    /// @notice Remove method signatures(methodIDs) from the whitelist.
+    /// @param _methods method signatures to be removed.
+    function removeMethodIds(bytes4[] calldata _methods) external onlyAdmin {
+        for (uint i = 0; i < _methods.length; i++) {
+            bytes4 method = _methods[i];
+            if (_methodIdWhitelist[method]) {
+                _methodIdWhitelist[method] = false;
+                emit RemovedMethodId(method);
+            }
+        }
+    }
+
+    /// @notice Add exclusive methods supported for a specific token that are not in the general whitelist.
+    /// @param _token token address.
+    /// @param _methods method signatures to be added.
+    function addExclusiveMethods(address _token, bytes4[] calldata _methods) external onlyAdmin {
+        for (uint i = 0; i < _methods.length; i++) {
+            bytes4 method = _methods[i];
+            require(_tokenInfoMap[_token].available, "non-existing token");
+            if (!(_methodIdWhitelist[method] || _tokenInfoMap[_token].exclusiveMethods[method])) {
+                _tokenInfoMap[_token].exclusiveMethods[method] = true;
+                emit AddedExclusiveMethod(_token, method);
+            }
+        }
+    }
+
+    /// @notice Remove exclusive methods supported for a specific token.
+    /// @param _token token address.
+    /// @param _methods method signatures to be removed.
+    function removeExclusiveMethods(address _token, bytes4[] calldata _methods) external onlyAdmin {
+        for (uint i = 0; i < _methods.length; i++) {
+            bytes4 method = _methods[i];
+            require(_tokenInfoMap[_token].available, "non-existing token");
+            if (_tokenInfoMap[_token].exclusiveMethods[method]){
```

Comment:
same here ... think removing not existing method should end up in a `revert` ...